src/store/pages.js

```diff
@@ -763,5 +778,37 @@ export default {
 			const response = await axios.get(getters.backlinksUrl(page.parentId, page.id))
 			commit(SET_BACKLINKS, { pages: response.data.data })
 		},
+
+		/**
+		 * Init full width store from browser storage
+		 *
+		 * @param {object} store the vuex store
+		 * @param {Function} store.commit commit changes
+		 */
+		[INIT_FULL_WIDTH_PAGEIDS]({ commit }) {
+			commit(SET_FULL_WIDTH_PAGEIDS, JSON.parse(persistentStorage.getItem('fullWidthPageIds') ?? '[]'))
+		},
+
+		/**
+		 * Set full width for a page
+		 *
+		 * @param {object} store the vuex store
+		 * @param {Function} store.commit commit changes
+		 * @param {object} store.getters getters of the store
+		 * @param {boolean} fullWidthView Whether full width view is enabled or not
+		 */
+		[SET_FULL_WIDTH_VIEW]({ commit, getters }, fullWidthView) {
+			const pageId = getters.currentPage.id
+			const fullWidthPageIds = JSON.parse(persistentStorage.getItem('fullWidthPageIds') ?? '[]')
+			if (fullWidthView && !fullWidthPageIds.includes(pageId)) {
+				fullWidthPageIds.push(pageId)
+				commit(SET_FULL_WIDTH_PAGEIDS, fullWidthPageIds)
+				persistentStorage.setItem('fullWidthPageIds', JSON.stringify(fullWidthPageIds))
```

Comment:
Would be nice to prefix this with the app id, considering we may want to share this state with text at some point maybe we can already use something like `text-fullWidthPageIds`?